### PR TITLE
Readds the stimball reaction + radball fire cleanup

### PR DESCRIPTION
**Pull request by vincentiusvin**
### About the pull request
Pseudo-port of Yog's improved stimballs: [#9510](https://github.com/yogstation13/Yogstation/pull/9510). 
Math got changed for balance and because i feel the old one was lackluster. Feel free to give input on this.
Due to balance concerns and how much people don't like seeimingly random sources of radiation, this reaction produces a radball that does not deal radiation but 20 cellular damage, still armorflagged as RAD.

Killed off the atom level `fire_nuclear_particle()` proc, made it into a proc for the HFR and the plain implementation for the supermatter.
Had to universalize the HFR nuclear particle firing because of this. Nothing major really.

### Why is it good for the game?
More content for atmos players; more usage for stimulum; a slightly more in depth reaction to our boring roster.

### Changelog
🆑
add: Re-adds the stimball reaction, angle calculation ported from Yogs.
code: Changed radball firing proc, small behavior change for the angle calculation on the HFR.
/🆑

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
We removed it due to bloat, mind selling me on why it's interesting/used/unique?

**Comment by vincentiusvin** [comment]
Like the old stimball, this will allow traitor atmostechs to create canisters or pipenets that shoot balls, giving them an interesting way to antagonize the station without resorting to plasmaflooding, fusion heat flooding, or explosives. This reaction works really good as an area denial weapon, but it can also work when hooked to distro, albeit not very well. The main appeal (for me at least) is on the way the reaction rate is counted, presenting an extra challange for techs to optimize their mix and make as much balls with as little gas as possible.

**Comment by LemonInTheDark** [comment]
How often do you think it'll get used?

**Comment by vincentiusvin** [comment]
To be honest, i don't know. The old stimball wasnt used either because the angle math broke and/or people dont know about it. I certainly hope that if this PR sees the light of day, the few atmos that knew about the reaction would use it and showcase how destructive/gimmicky/cool it is and inspire other players to use it.
That is the problem however, a destructive reaction certainly inspire others to use it, however people have pointed out that radiation isn't fun to deal with. Nerfing the balls to deal cloneloss surely reduce the appeal for atmos players to pursue the reaction, but i hope that the way i implemented it should be enough to strike a balance between destruction and also balance.

**Comment by Jared-Fogle** [comment]
Lemon implied in Discord he didn't want this, and our PR list is getting massive. He can reopen this if he wants it.

**Comment by LemonInTheDark** [comment]
I'm more alright with this now that we're not bound to reaction count for the cost of react(). Might need to kill it off at a later date, but if I do that I'm taking a lot of other stuff with it, so let's jive for now

**Comment by vincentiusvin** [comment]
Thought i could rename it. Ugh.

**Comment by LemonInTheDark** [comment]
rip

**Comment by vincentiusvin** [comment]
Should be ready for review. At the current rate the reaction can make a maximum amount of 12 balls per tick, with the most efficient run costing 2 moles per ball. Running efficiently, this would mean a 10k stim-trit mix can make roughly 5000 balls, spread over 834 ticks. The projectile still deals 20 damage per shot. Do you think this is balanced enough?

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 1]
```diff
@@ -551,6 +555,7 @@
 		/datum/gas/tritium = 30,
 		/datum/gas/bz = 20,
 		/datum/gas/nitryl = 30,
+		/datum/gas/plasma = MINIMUM_MOLE_COUNT,
```
Comment (Ghilker) [inline code/modules/atmospherics/gasmixtures/reactions.dm:558]: why you added plasma as requirement?

Comment (vincentiusvin) [reply code/modules/atmospherics/gasmixtures/reactions.dm:558]: It has always been a requirement, just not explicitly. The plasma # is hooked to a min() proc for the moles formed. I thought that formally putting it there might suit the new react system better, thought i didn't give much thought into it.

**Comment by vincentiusvin** [comment]
On reflection, I don't think this will be a good addition. Cloneloss particles can just be dodged by resting, making this reaction useful only for trapping new players. Making it a radiation particle will just make it too powerful and fun will be had at the expense of the crew at large. I'm closing this for good, but I'll atomize the smaller changes alongside it like the stimball requirement thing.

**Pull request closed without merging**